### DISCUSSION TITLE
Creating Unique Constraint without Index

### DISCUSSION THREAD

**Discussion by Jeyanthinath** [pocketbase/pocketbase#2932] [Q&A] [answered]
From what I saw in the settings, there is an only possibility where I need to index the row when I need to make it unique. 

Does Pocketbase support makes the field unique without adding index to the field ?

**Comment by ganigeorgiev** [comment]
I don't understand what do you mean by _"where I need to index the row when I need to make it unique"_, but there is no option at the moment to create a unique constraint without index since it is handled on DB level (_before v0.14.0 it was possible to define a unique constraint on application level as field validator but it had other limitations such as lack of combined constraints and expressions_).

For most use cases it shouldn't really matter. Does the unique index has that much impact on your inserts?

**Reply by Lex-2008** [reply, marked as answer]
Worth noting that in SQLite, [in most cases, UNIQUE constraints are implemented by creating a unique index in the database][x].

[x]: https://www.sqlite.org/lang_createtable.html#unique_constraints

Also, mentioning performance - usually, UNIQUE constraint means that during INSERT operation you want to check that the table doesn't already have a value you want to insert. Without an index, this would probably mean scanning whole table. And I believe that in most cases it's worse than adding an entry to an index :)

**Reply by Jeyanthinath** [reply]
Thanks @ganigeorgiev @Lex-2008 for the answers.

Initially I worried about putting extra write load to the system as SQLite is not super powerful in handling writes. Now I am clarified on why we need an index for the unique constraint field.
